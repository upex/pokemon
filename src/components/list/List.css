.pokemon-container {
  padding: 100px 10px 60px 10px;
  display: flex;
  align-items: center;
  justify-content: center;
}