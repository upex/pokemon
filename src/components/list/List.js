import React from 'react';
import PropTypes from 'prop-types';
import './List.css';
import Pokemon from '../pokemon/Pokemon';

function List(props) {
  let { items } = props;
  return (
    <div className="pokemon-container">
      {items.map((monster, index) =>
        <Pokemon key={monster.name} id={index + 1} url={monster.url} pokemon={monster.name} />
      )}
    </div>
  );
}
List.propTypes = {
  items: PropTypes.array,
};
export default List;