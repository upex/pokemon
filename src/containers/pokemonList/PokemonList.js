import React, { Component } from 'react';
import axios from 'axios';
import List from '../../components/list/List';
import Pagination from '../../components/pagination/Pagination';

class PokemonList extends Component {
  _isMounted = false;
  constructor(props) {
    super(props);
    this.state = {
      pokemon: [],
      currentPage: 1,
      pokemonPerPage: 3
    };
  }

  getPokemon() {
    axios.get(`https://pokeapi.co/api/v2/pokemon/?limit=300`)
    .then(res => {
      if (this._isMounted) {
        this.setState({
          pokemon: res.data.results
        });
      }
    });
  }

  componentDidMount() {
    this._isMounted = true;
    this.getPokemon();
  }

  handlePrevious = () => {
    this.setState({
      currentPage: this.state.currentPage - 1
    });
  }

  handleNext = () => {
    this.setState({
      currentPage: this.state.currentPage + 1
    });
  }
  componentWillUnmount() {
    this._isMounted = false;
  }
  render() {
    let { pokemon, currentPage, pokemonPerPage } = this.state;
    const begin = currentPage * pokemonPerPage;
    const end = begin - pokemonPerPage;
    const currentPokemon = pokemon.slice(end, begin);

    return (
      <>
        <List items={currentPokemon} />
        <Pagination
        currentPage={this.state.currentPage}
        pokemonLength={pokemon.length}
        handlePrevious={this.handlePrevious}
        handleNext={this.handleNext} />
      </>
    );
  }
}

export default PokemonList;