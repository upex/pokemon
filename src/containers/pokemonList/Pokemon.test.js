import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import PokemonList from './PokemonList';
import List from '../../components/list/List';
Enzyme.configure({ adapter: new Adapter() });

it('renders list without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<PokemonList />, div);
  ReactDOM.unmountComponentAtNode(div);
});
describe("PokemonList component", () => {
  test("renders", () => {
    const items = [
      {
        name:"bulbasaur",
        url:"https://pokeapi.co/api/v2/pokemon/1/"
      },
      {
        name:"ivysaur",
        url:"https://pokeapi.co/api/v2/pokemon/2/"
      },
      {
        name:"venusaur",
        url:"https://pokeapi.co/api/v2/pokemon/3/"
      }
    ];
    const wrapper = shallow(<List items={items} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});