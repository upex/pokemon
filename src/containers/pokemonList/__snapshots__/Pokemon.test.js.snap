// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList component renders 1`] = `ShallowWrapper {}`;
